order/src/main/java/com/yas/order/service/OrderService.java

```diff
@@ -1,18 +1,73 @@
 package com.yas.order.service;
 
+import com.yas.order.model.Order;
+import com.yas.order.model.OrderItem;
+import com.yas.order.repository.OrderItemRepository;
 import com.yas.order.repository.OrderRepository;
+import com.yas.order.viewmodel.OrderPostVm;
+import com.yas.order.viewmodel.OrderVm;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.apache.commons.lang3.Validate;
+
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Slf4j
 @Service
 @Transactional
 public class OrderService {
     private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+
 
-    public OrderService(OrderRepository orderRepository) {
+    public OrderService(OrderRepository orderRepository, OrderItemRepository orderItemRepository) {
         this.orderRepository = orderRepository;
+        this.orderItemRepository = orderItemRepository;
     }
 
+
+    public OrderVm createOrder(OrderPostVm orderPostVm) {
+        Validate.notNull(orderPostVm, "Order cannot be null");
+        Validate.notNull(orderPostVm.orderItemPostVms(), "Order Items cannot be null");
```

Comment:
You can use these annotations in the controller and record vm to validate these fields. Don't necessary to validate in the Services layer.

- Add annotation @Validated to Rest Controller class
- Add @Valid @NotNull before OrderPostVm params
- In record OrderPostVM you can also add @NotNull before OrderItemPostVms

Reply:
thanks